lua/nvim-tree/keymap.lua

```diff
@@ -1,5 +1,24 @@
 local M = {}
 
+--- Apply mappings to a scratch buffer and return buffer local mappings
+--- @param fn function(bufnr) on_attach or default_on_attach
+--- @return table as per vim.api.nvim_buf_get_keymap
+local function generate_keymap(fn)
+  -- create a scratch buffer
+  local scratch_bufnr = vim.api.nvim_create_buf(true, true)
```

Comment:
The parameters here should be `false, true`, as the first parameters is `listed` which sets the `buflisted` option. If we're just using it in the background to get keymaps, it does not need to be in the user's buffer list, especially if we're deleting it right away anyway.

Reply:
Thank you so much! Fixed.